### PR TITLE
inserting kmeans script in clean script with kmeans notebook

### PR DESCRIPTION
**Pull request by LDB-001**

### CODE REVIEW

File: src/data/clean_dataset.py [hunk 1]
```diff
@@ -133,6 +133,74 @@ def clean_strucute(df):
...
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['IDOTRR'],"16")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['ClearCr'],"17")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['StoneBr'],"18")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Blmngtn'],"19")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['SWISU'],"20")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['MeadowV'],"21")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['BrDale'],"22")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Veenker'],"23")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NPkVill'],"24")
+    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Blueste'],"25")
```
Comment (mendesfabio) [inline src/data/clean_dataset.py]: You are repeating yourself here. With lists and loops, you could reduce it. Take a look at this [simple example](https://realpython.com/lessons/dry-principle/).

File: src/data/clean_dataset.py [hunk 2]
```diff
@@ -133,6 +133,74 @@ def clean_strucute(df):
...
+    df_k=df[['Neighborhood','SalePrice']]
+    
+    import numpy as np
+    from sklearn.datasets.samples_generator import (make_blobs,
+                                                    make_circles,
+                                                    make_moons)
+    from sklearn.cluster import KMeans, SpectralClustering
+    from sklearn.preprocessing import StandardScaler
+    from sklearn.metrics import silhouette_samples, silhouette_score
+    from warnings import filterwarnings
```
Comment (mendesfabio) [inline src/data/clean_dataset.py]: It's not Pythonic import libs inside functions. Move it to the top of the file.

File: src/data/clean_dataset.py [hunk 3]
```diff
@@ -133,6 +133,74 @@ def clean_strucute(df):
...
+
+    df['Cluster Neighborhood'] = clusters
+    df.drop('Neighborhood',axis=1)
+
+    return df 
+
+
 def clean(df):
     '''
     Script to clean a entiry dataset.
```
Comment (mendesfabio) [inline src/data/clean_dataset.py]:
```suggestion
    Script to clean an entire dataset.
```

File: notebooks/Neighborhood applied with kmeans.ipynb [hunk 4]
```diff
@@ -0,0 +1,1399 @@
+{
```
Comment (yvesfracari) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>train = pd.read_csv('../data/raw/train.csv')</p><p> </p><p>and fix the notebook name!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (yvesfracari) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>If you are not using the packages delete the lines!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (yvesfracari) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>You already imported matplotlib</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (yvesfracari) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>Check the FSN comment in the another file and fix here too!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (yvesfracari) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>You already imported kmeans!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (yvesfracari) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>English plz!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (yvesfracari) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>Nice notebook! Pay more attention in imports to don't repeat yourself!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (LDB-001) [reply notebooks/Neighborhood applied with kmeans.ipynb]: <p>Sure!</p>

Comment (LDB-001) [reply notebooks/Neighborhood applied with kmeans.ipynb]: <p>done!</p>

Comment (LDB-001) [reply notebooks/Neighborhood applied with kmeans.ipynb]: <p>done!</p>

Comment (LDB-001) [reply notebooks/Neighborhood applied with kmeans.ipynb]: <p>where did i repeat?</p>

File: src/data/clean_dataset.py (hunk 1 above)
Comment (LDB-001) [reply src/data/clean_dataset.py]: done!

File: notebooks/Neighborhood applied with kmeans.ipynb (hunk 4 above)
Comment (Stefenon) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>The Neighborhood category names are scrambled on x axis. Although it's not the most important thing of the notebook, it would be great to fix it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (Stefenon) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>The category names are scrambled here too.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (Stefenon) [inline notebooks/Neighborhood applied with kmeans.ipynb]: <p>Is the objective of this chart to explain the K-Means method? If yes, I think you could make a brief explanation about it in a markdown. Otherwise, I believe it would be better to leave it out of the notebook, as it's not related to our dataset.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Comment by Stefenon** [comment]
<p>Nice job with the data cleaning! I believe this is probably gonna improve our model quite a lot.</p><p> </p><p>One thing I believe you should pay more attention is to the documentation of the notebook. Me, as a "K-Means noob", didn't really understand what was going on most of the notebook hahahah. </p><p> </p><p>Maybe you could explain a little bit what your code is doing, especially those parts where you use sci-kit learn features, since they are not that common.</p>

File: notebooks/Neighborhood applied with kmeans.ipynb (hunk 4 above)
Comment (yvesfracari) [reply notebooks/Neighborhood applied with kmeans.ipynb]: <p>Two cells above</p>

File: src/data/clean_dataset.py [hunk 5]
```diff
@@ -147,36 +147,12 @@ def kmeans (df):
...
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['MeadowV'],"21")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['BrDale'],"22")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Veenker'],"23")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NPkVill'],"24")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Blueste'],"25")
-
-    df_k['Neighborhood'] = df_k['Neighborhood'].astype(int)
-
+    for index, value in enumerate(lista):
+        df_k['Neighborhood'] = df_k['Neighborhood'].replace(lista, index)
```
Comment (Stefenon) [inline src/data/clean_dataset.py]: Did this work properly? Maybe I'm wrong, but I believe it should be `replace([value], f'{index}')`.

File: notebooks/4.0-ldb-neighborhood-applied-with-kmeans.ipynb [hunk 6]
```diff
@@ -0,0 +1,1082 @@
+{
```
Comment (yvesfracari) [inline notebooks/4.0-ldb-neighborhood-applied-with-kmeans.ipynb:1]: <p>You are repeating some imports, I suggest you to create a cell (the first code one) just with imports!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/neo-empresarial/house_pricing_challenge/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: src/data/clean_dataset.py [hunk 7]
```diff
@@ -1,6 +1,13 @@
...
 import os, sys
-
+import numpy as np
+    from sklearn.datasets.samples_generator import (make_blobs,
+                                                    make_circles,
+                                                    make_moons)
+    from sklearn.cluster import KMeans, SpectralClustering
+    from sklearn.preprocessing import StandardScaler
+    from sklearn.metrics import silhouette_samples, silhouette_score
+    from warnings import filterwarnings
```
Comment (yvesfracari) [inline src/data/clean_dataset.py]: Take care with indentation!
```suggestion
from sklearn.datasets.samples_generator import (make_blobs,
                                                make_circles,
                                                make_moons)
from sklearn.cluster import KMeans, SpectralClustering
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import silhouette_samples, silhouette_score
from warnings import filterwarnings
```
